torch/csrc/autograd/engine.cpp

```diff
@@ -503,6 +555,24 @@ struct ClearCallbacks {
   std::mutex& callbacks_lock;
 };
 
+struct LocalBackwardGuard {
+  LocalBackwardGuard(bool run_parallel)
+    : run_parallel(run_parallel) {
+    if (!run_parallel) {
+      use_local_queue++;
+      local_queue = std::make_shared<ReadyQueue>();
```

Comment:
How does this work with nested `LocalBackwardGuard`? It looks like nested LocalBackwardGuard with `run_parallel=false` will stomp the `local_queue` shared_ptr.

Reply:
Ugh, that's obviously a bug! It should be guarded by a check that it's not defined yet. I had it, but apparently I accidentally got rid of it.